src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [modalLoading, setModalLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setModalLoading(true);
+
+    if (!currentTodo) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(data => setUser({ ...data }))
+      .finally(() => setModalLoading(false));
+  }, []);
```

Comment:
```suggestion
  }, [currenctTodo]);
```
Just add it to the dependencies list)

![image](https://github.com/user-attachments/assets/73f28fa3-f422-40d4-a347-c8f5f86d25e8)